lib/msal-common/src/config/ClientConfiguration.ts

```diff
@@ -46,7 +46,8 @@ export type ClientConfiguration = {
     telemetry?: TelemetryOptions,
     serverTelemetryManager?: ServerTelemetryManager | null,
     persistencePlugin?: ICachePlugin | null,
-    serializableCache?: ISerializableTokenCache | null
+    serializableCache?: ISerializableTokenCache | null,    
+    appTokenProvider?: IAppTokenProvider | null
```

Comment:
Since this is only relevant to the `ClientCredentialClient` I'd propose including this as a parameter on the `ClientCredentialClient` constructor instead of polluting all of our client classes

Reply:
We do not provide "ClientCredentialClient" specific config, we only control requests that way which I guess why Bogdan chose config.

Reply:
I'm suggesting to add it as an optional parameter on [this line](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/867516c11c83639890abe2fcd6fc6e73ed659644/lib/msal-common/src/client/ClientCredentialClient.ts#L29)

Reply:
@tnorling - I am not sure I understand how the app developer can use object? I was working based on this sample: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/42287f399ea292b4b25eb689eb4b304b30646d62/samples/msal-node-samples/client-credentials/index.js#L77